src/impl.ml

```diff
@@ -673,7 +673,8 @@ let write_stream common s destination source_protocol destination_protocol preze
   let use_ssl = match endpoint with Https _ -> true | _ -> false in
   ( match endpoint with
     | File path ->
-      Lwt_unix.openfile path [ Unix.O_RDWR; Unix.O_CREAT ] 0o0644 >>= fun fd ->
+      let unix_fd = Vhd_format_lwt.File.openfile path true 0o644 in
```

Comment:
What happens to O_CREAT here, and is it relevant?

Reply:
In our case the file has already been created. It's possible we might want to follow the pattern later in the file: https://github.com/jonludlam/vhd-tool/blob/f039b64a025fd546519cc434bc6033d38509b54b/src/impl.ml#L968-L972

Reply:
Sounds good, factor out that out into an `open_odirect_rw` function?

Reply:
I can't quite use the same function - that does `Vhd_format_lwt.IO.openfile` rather than a `Vhd_format_lwt.File.openfile` - the former returns an opaque type encapsulating the fd that I don't think I can break apart.

Reply:
Oh!, my mistake, there it is:

https://github.com/xapi-project/ocaml-vhd/blob/master/vhd_format_lwt/iO.mli#L22

Reply:
From the diff it is not obvious that `O_DIRECT` is now being used as the commit message suggests. I think this needs to explained either in the commit message or in a comment.

Reply:
it's awkward because the o_directness is dictated by a global var, so it's be a bit of a misnomer. I've named it 'open_with_creat' instead.

@lindig I'll update the commit message when I squash these after everyone is happy with the code.